Lib/fontTools/feaLib/builder.py

```diff
@@ -705,19 +705,11 @@ def set_language(self, location, language, include_default, required):
         self.cur_lookup_ = None
 
         key = (self.script_, language, self.cur_feature_name_)
-        if not include_default:
-            # don't include any lookups added by script DFLT in this feature
+        lookups = self.features_.get((key[0], 'dflt', key[2]))
+        if language != 'dflt' and include_default and lookups:
+            self.features_[key] = lookups[:]
+        else:
             self.features_[key] = []
-        elif language != 'dflt':
-            # add rules defined between script statement and its first following
-            # language statement to each of its explicitly specified languages:
-            # http://www.adobe.com/devnet/opentype/afdko/topic_feature_file_syntax.html#4.b.ii
-            lookups = self.features_.get((key[0], 'dflt', key[2]))
-            dflt_lookups = self.features_.get(('DFLT', 'dflt', key[2]), [])
-            if lookups:
-                if key[:2] in self.get_default_language_systems_():
-                    lookups = [l for l in lookups if l not in dflt_lookups]
-                self.features_.setdefault(key, []).extend(lookups)
```

Comment:
In the previous code, the occurrence of a non-dflt language statement (with include_dflt implicitly true) would _extend_ the lookup list of the feature record for that script/language combo with the current script's default lookups (i.e. those between the script statement and the first non-dflt language statement). And the occurence of a `language dflt` (either explicitly or implicitly via a script statement) would do nothing, besides setting the current `self.language_systems` to this specific script/dflt (replacing the global set from languagesystem statements). 

Now, with your code, the occurrence of a non-dflt (including) language statement sets the feature's lookup list for this script/language to the script's default lookups, discarding any feature-level default lookups (i.e. occurring before any script statement). Moreover with your patch, when a `language dflt` statement is seen , the current script/language lookup list is always emptied.

The result of these modifications can be seen in the adjusted test results in `spec4h1.ttx`, in which lookup 1 (the one with the f-ligatures rules defined at the top of liga feature before any script statement) is no longer included in the lookup lists of the feature records of the latn/dflt, latn/DEU and latn/TRK.

I believe this is a deviation from the current spec.